ci/official/utilities/setup.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Common setup for all TF scripts.
+#
+# Make as FEW changes to this file as possible. It should not contain utility
+# functions (except for tfrun); use dedicated scripts instead and reference them
+# specifically. Use your best judgment to keep the scripts in this directory
+# lean and easy to follow. When in doubt, remember that for CI scripts, "keep it
+# simple" is MUCH more important than "don't repeat yourself."
+
+# -e: abort script if one command fails
+# -u: error if undefined variable used
+# -x: log all commands
+# -o pipefail: entire command fails if pipe fails. watch out for yes | ...
+# -o history: record shell history
+# -o allexport: export all functions and variables to be available to subscripts
+set -euxo pipefail -o history -o allexport
+
+# Import all variables as set in $TFCI, which should be a file like those in
+# the envs directory that sets all TFCI_ variables, e.g. /path/to/envs/local_cpu
+source "$TFCI"
```

Comment:
Should we attempt to preserve any envs that are set before this point?  I could see things like `TFCI_UPLOAD_WHL_PYPI_ARGS` or `TFCI_DOCKER_ENABLE` being nice to have preset or a user being initially confused as to why just setting one of the envs before the script doesn't work.  I could see the opposite though where that could be confusing that a replacement doesn't happen.

Reply:
Yes, good idea. I adjusted this such that TFCI is only sourced if specified. That way, someone can e.g. source one of the env files, then overwrite it with a follow-up "export TFCI_DOCKER_ENABLE=1" for example. We'll have to be careful to document it more than it is right now.